parsec/core/gui/new_version.py

```diff
@@ -20,78 +22,95 @@
-            else:
-                return latest_from_head, None
-
-    latest_from_head, json_releases = await trio.to_thread.run_sync(_fetch_json_releases)
-    if json_releases:
-        current_arch = QSysInfo().currentCpuArchitecture()
-        if current_arch == "x86_64":
-            win_version = "win64"
-        elif current_arch == "i386":
-            win_version = "win32"
-        else:
-            return latest_from_head, url
+class ApiReleaseAssetSchema(BaseSchema):
+    name = fields.String(required=True)
+    browser_download_url = fields.String(required=True)
+
+
+VersionField = fields.str_based_field_factory(Version)
+
+
+class ApiReleaseSchema(BaseSchema):
+    draft = fields.Boolean(missing=False)
+    prerelease = fields.Boolean(missing=False)
+    version = VersionField(required=True)
+    assets = fields.List(fields.Nested(ApiReleaseAssetSchema), missing=[])
+
 
-        latest_version = Version("0.0.0")
-        latest_url = ""
+class ApiReleasesSchema(BaseSchema):
+    releases = fields.List(fields.Nested(ApiReleaseSchema), missing=[])
 
+
+api_releases_serializer = JSONSerializer(ApiReleasesSchema)
+
+
+async def fetch_json_releases(api_url: str) -> Optional[List]:
+    def _do_http_request():
         try:
-            for release in json_releases:
-                if release["draft"]:
-                    continue
-                if release["prerelease"] and not check_pre:
-                    continue
-                for asset in release["assets"]:
-                    if asset["name"].endswith(f"-{win_version}-setup.exe"):
-                        asset_version = _extract_version(release["tag_name"])
-                        if (
-                            asset_version
-                            and asset_version > latest_version
-                            and (not asset_version.is_prerelease or check_pre)
-                        ):
-                            latest_version = asset_version
-                            latest_url = asset["browser_download_url"]
-        # In case something went wrong, still better to redirect to GitHub
-        except (KeyError, TypeError):
-            return latest_from_head, url
-        if latest_version > current_version:
-            return latest_version, latest_url
-    elif latest_from_head > current_version:
-        # There is a new release flagged as stable on GitHub, but we failed to load the json
-        return latest_from_head, url
+            with urlopen(Request(api_url, method="GET")) as req_api:
+                return json.loads(req_api.read())
+        except JSONDecodeError as exc:
+            logger.error("Cannot deserialize releases info from API", exc_info=exc, api_url=api_url)
+            return None
+        except HTTPError:
+            return None
+
+    return await trio.to_thread.run_sync(_do_http_request)
+
+
+async def do_check_new_version(
+    api_url: str, allow_prerelease: bool = False
```

Comment:
Deleting the `url` argument causes errors elsewhere making the pop-up not show at all

```
File "parsec/core/gui/trio_thread.py", line 95, in _run_fn
    result = await self._fn(*self._args, **self._kwargs)
TypeError: do_check_new_version() got an unexpected keyword argument 'url'
```